mount/mount_unix.go

```diff
@@ -21,7 +21,7 @@ func Mount(device, target, mType, options string) error {
 
 // Unmount lazily unmounts a filesystem on supported platforms, otherwise does
 // a normal unmount. If target is not a mount point, no error is returned.
-func Unmount(target string) error {
+var Unmount = func(target string) error {
```

Comment:
I'd rather not do that for the sake of a single unit test.

Unfortunately I don't know of any other way to make the deferred unmount fail.

Reply:
Hmm, maybe I do.

Try to create a mount which is shadowed by another mount. This will be shown in mountinfo, but the directory will not be there, so the unmount will fail with ENOENT.

IOW something like
1. mount "tmp/dir1/subdir" using tmpfs.
2. mount "tmp/dir1" using tmpfs.

Now, RecursiveUnmount will try to unmount "tmp/dir1/subdir", and since it's hidden by the second mount, its unmount will fail with ENOENT.

Reply:
Thanks for the suggestion. I did not know about shadow mounts. I updated the test by using your suggestion.